Resources/public/js/views/services/plugins/ez-locationsearchplugin.js

```diff
@@ -222,31 +223,41 @@ YUI.add('ez-locationsearchplugin', function (Y) {
          * @param {Array|Null} locationStructArr
          * @param {Function} callback
          */
-        _loadContentForLocations: function (locationStructArr, callback) {
-            var tasks = new Y.Parallel(),
-                loadContentError = false;
+        _loadContentForLocations: function (viewName, locationStructArr, callback) {
+            var contentService = this.get('host').get('capi').getContentService(),
+                contentIdCriterions = '',
+                loadContentError = false,
+                that = this,
+                query;
 
             Y.Array.each(locationStructArr, function (locationStruct, i) {
-                var content = new Y.eZ.Content(
-                        {id: locationStruct.location.get('contentInfo').get('id')}
-                    ),
-                    capi = this.get('host').get('capi'),
-                    options = {api: capi},
-                    endContentLoad = tasks.add(function (error, response) {
-                        if (error) {
-                            loadContentError = true;
-                            return;
-                        }
-
-                        locationStructArr[i].content = content;
-                    });
-
-                content.load(options, endContentLoad);
+                contentIdCriterions =
+                    contentIdCriterions +
+                    "," +
+                    locationStruct.location.get('contentInfo').get('contentId');
             }, this);
 
-            tasks.done(function () {
-                callback(loadContentError, locationStructArr);
+            query = contentService.newViewCreateStruct('contents-loading-'+ viewName, 'ContentQuery');
+            query.body.ViewInput.ContentQuery.Criteria = {
+                "ContentIdCriterion": contentIdCriterions
+            };
+
+            contentService.createView(query, function (err, response) {
+                if (err) {
+                    loadContentError = true;
+                    callback(loadContentError, locationStructArr);
+                    return;
+                }
+                Y.Array.each(response.document.View.Result.searchHits.searchHit, function (hit, i) {
+                    var Content = that.get('contentModelConstructor'),
+                        content;
+
+                    content = new Content();
+                    content.loadFromHash(hit.value.Content);
+                    locationStructArr[i].content = content;
+                });
             });
+            callback(loadContentError, locationStructArr);
```

Comment:
this call should be in the `createView` callback, otherwise you will execute the callback before the Content items are actually loaded and added to `locationStructArr`.